src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -1098,3 +1124,60 @@ func isRedpandaClusterVersionManaged(
 	}
 	return true
 }
+
+func patchStatus(ctx context.Context, c client.Client, observedCluster *vectorizedv1alpha1.Cluster, mutator func(cluster *vectorizedv1alpha1.Cluster)) (vectorizedv1alpha1.ClusterStatus, error) {
+	name := types.NamespacedName{
+		Namespace: observedCluster.Namespace,
+		Name:      observedCluster.Name,
+	}
+	cluster := &vectorizedv1alpha1.Cluster{}
+	err := c.Get(
+		ctx,
+		name,
+		cluster,
+	)
+	if err != nil {
+		return vectorizedv1alpha1.ClusterStatus{}, err
+	}
+
+	clusterPatch := client.MergeFrom(cluster.DeepCopy())
+	mutator(cluster)
+
+	err = c.Status().Patch(ctx, cluster, clusterPatch)
```

Comment:
nit: prefer inline error handling to avoid polluting the name `err` when possible. It also helps readers understand that `err` is only considered within the scope of the `if` and isn't intended to live further.